homework/list.js

```diff
@@ -1,79 +1,95 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+    var TodoListManager = function () { //constructor
+        this._listNoteContent = [];
+        this._wrapper = document.querySelector('#note-list-wrapper');
+    }
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+    TodoListManager.prototype = { //its prototype
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+        start() {
+                 this.fetchList().then(function(data) {
```

Comment:
Good. Although you should try to compose the chain with multiple `then` and `catch`. And you should note that the `catch` should **log AND re-throw** the error, to prevent the following chains keep the error in silent and perform some unexpected effects.